homeassistant/components/home_connect/switch.py

```diff
@@ -35,18 +81,82 @@ async def async_setup_entry(
     def get_entities():
         """Get a list of entities."""
         entities = []
-        hc_api = hass.data[DOMAIN][config_entry.entry_id]
+        hc_api: ConfigEntryAuth = hass.data[DOMAIN][config_entry.entry_id]
         for device_dict in hc_api.devices:
             entity_dicts = device_dict.get(CONF_ENTITIES, {}).get("switch", [])
             entity_list = [HomeConnectProgramSwitch(**d) for d in entity_dicts]
             entity_list += [HomeConnectPowerSwitch(device_dict[CONF_DEVICE])]
             entity_list += [HomeConnectChildLockSwitch(device_dict[CONF_DEVICE])]
+            # Auto-discover entities
+            entities += [
+                HomeConnectSwitch(
+                    device=device_dict[CONF_DEVICE], entity_description=description
+                )
+                for description in SWITCHES
+                if description.exist_fn(device_dict[CONF_DEVICE])
+            ]
             entities += entity_list
         return entities
 
     async_add_entities(await hass.async_add_executor_job(get_entities), True)
 
 
+class HomeConnectSwitch(HomeConnectEntity, SwitchEntity):
+    """Generic switch class for Home Connect Binary Settings."""
+
+    def __init__(
+        self,
+        device: HomeConnectDevice,
+        entity_description: HomeConnectSwitchEntityDescription,
+    ) -> None:
+        """Initialize the entity."""
+        self.entity_description: HomeConnectSwitchEntityDescription = entity_description
+        super().__init__(device=device, desc=self.entity_description.key)
+        self._key = self.entity_description.on_key
+        self._attr_available = False
```

Comment:
I think we can remove the constructor if you replace all `self._key` with `self.entity_description.key` and set `_attr_available` outside of the constructor